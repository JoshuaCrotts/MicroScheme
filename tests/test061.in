(define abs (lambda (x) (if (<= x 0) (* x -1) x)))
(define square (lambda (x) (* x x)))

(define (average x y)
  (/ (+ x y) 2))
(define (improve guess x)
  (average guess (/ x guess)))
(define (good-enough? guess x)
  (< (abs (- (square guess) x)) 0.001))
(define (sqrt-iter guess x)
  (if (good-enough? guess x)
      guess
      (sqrt-iter (improve guess x)
                 x)))
(define (my-sqrt x)
  (sqrt-iter 1.0 x))

(my-sqrt 9)
(my-sqrt (+ 100 37))
(my-sqrt (+ (my-sqrt 2) (my-sqrt 3)))
(square (my-sqrt 1000))
