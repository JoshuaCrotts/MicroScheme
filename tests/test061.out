3.000091554131380178530556191348134
11.70469991775814352594298683199054
1.773927902320789248871042252700405
1000.00036992436602947805607554362322337393225463327334497336576501760000000000000000000000